crates/astria-composer/src/executor/tests.rs

```diff
@@ -480,3 +538,33 @@ async fn two_seq_actions_single_bundle() {
         );
     }
 }
+
+/// Test to check that executor's configured sequencer chain id and sequencer's actual chain id
+/// match
+#[tokio::test]
+async fn should_exit_if_mismatch_sequencer_chain_id() {
```

Comment:
So this test is dangerous because it could silently break: somebody might break the chain-ID check in a refactor, but then subsequently add code that still causes this method to exit with an error. Then, under test conditions, everything looks peachy, even though we have a problem.

IMO the most straightforward way to fix this is:

1. refactor the check into a free-standing function like so:

```rust
#[derive(Debug, thiserror::Error)]
enum EnsureChainIdError {
    // <other error variants from getting genesis, etc>
    #[error("expected chain ID `{expected}`, but received `{actual}`)]
    WrongChainId { expected: String, actual: tendermint::chain::Id, }
}

fn ensure_chain_id_is_correct(client: tendermint::HttpClient, expected_chain_id: String) -> Result<(), EnsureChainIdError> {
    // code to get the chain ID
    if remote_chain_id != expected_chain_id {
        return Err(WrongChainId { expected: expected_chain_id, actual: remote_chain_id});
    }
    Ok(())
}
```

You can then actually check the function without setting up the rest, which makes things a bit easier, i.e. you could check it like so:
```rust
match err {
    EnsureChainIdError::WrongChainId { expected, actual } => {
        assert_eq!(expected, <configured-expected>);
        assert_eq!(actual, <configured-actual>);
    }
    other => panic!("expected `EnsureChainIdError::WrongChainId`, but got `{other}`),
}
```


If you still want to test `run_until_stopped` (like above, just some more code):

```rust
// <code as it's written right now>
let err = exector.run_until_stopped().await.expect_err("should exit with an error when reading a bad chain ID, but exited with success");

for cause in err.chain() {
    if let Some(err) = cause.downcast_ref::<EnsureChainIdError>() {
        match err {
            EnsureChainIdError::WrongChainId { expected, actual } => {
                assert_eq!(expected, <configured-expected>);
                assert_eq!(actual, <configured-actual>);
            }
            other => panic!("expected `EnsureChainIdError::WrongChainId`, but got `{other}`),
     }
}
```

Reply:
Resolved in 8523e04.